posthog/tasks/subscriptions.py

```diff
@@ -1,25 +1,42 @@
 import uuid
 from datetime import datetime, timedelta
+from time import sleep
 from typing import List, Optional
 
 import structlog
+from celery import group
 
-from posthog import settings
 from posthog.celery import app
 from posthog.email import EmailMessage
+from posthog.models.dashboard import Dashboard
+from posthog.models.dashboard_tile import DashboardTile
 from posthog.models.exported_asset import ExportedAsset
 from posthog.models.subscription import Subscription, get_unsubscribe_token
 from posthog.tasks.exporter import export_task
+from posthog.utils import absolute_uri
 
 logger = structlog.get_logger(__name__)
 
+MAX_ASSET_COUNT = 6
+UTM_TAGS = "utm_source=posthog&utm_medium=email&utm_campaign=subscription_report"
+
 
 def get_unsubscribe_url(subscription: Subscription, email: str) -> str:
-    return f"{settings.SITE_URL}/unsubscribe?token={get_unsubscribe_token(subscription, email)}"
+    return absolute_uri(f"/unsubscribe?token={get_unsubscribe_token(subscription, email)}&{UTM_TAGS}")
+
+
+def get_tiles_ordered_by_position(dashboard: Dashboard) -> List[DashboardTile]:
+    tiles = list(DashboardTile.objects.filter(dashboard=dashboard).order_by("insight__order").all())
```

Comment:
I think you need to `prefetch` or `select` `_related` here so that you don't load the insights one by one when iterating.

You can test whether that is true by writing a similar test to https://github.com/PostHog/posthog/blob/e5243a343042b6e590875a177646800d9e35447d/posthog/api/test/test_feature_flag.py#L815

or https://github.com/PostHog/posthog/blob/bb0872b7dd823b03fef5da111270730d0ee49e78/posthog/api/test/test_dashboard.py#L203

Check the number of queries is stable regardless of number of insights on dashboard

create several

Reply:
Also, do we want `.all()` or do we want six?

Reply:
Yep, I _think_ you could move `MAX_ASSET_COUNT` here to reduce what we're loading from DB

Reply:
I think you can use an annotation to get the count alongside. So you have a single (or at least minimal) queries and let the django ORM sort it out...

TBH just the select related would probably be enough to avoid adding slow query without over-pre-emptively-optimising :)